packages/nimble-components/src/table/specs/README.md

```diff
@@ -132,13 +132,17 @@ The various APIs/features of the `nimble-table` will be split up amongst several
     -   Define how the action menu gets associated with a particular column
     -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
 
+_Attributes_
+
+-   `id-field-name` - An optional string attribute that specifies the field name within a row's record to use as a row's ID. If the attribute is not defined or a field with the specified name does not exist in a row's record, a default ID will be created. The default ID is the index of the row's record within the table's `data` property. The value associated with the field name must be unique across all records in the table's data.
```

Comment:
Should we do api defaults magic and have the default resolved value be `id` for an undefined `id-field-name`?

- If the attribute is not defined:
  - If all (or first?) records have a field named id, then the field `id` is used and the id mode is 'field'
  - if the records do not have a field named id, then the id mode is 'index'
- If the attribute is defined:
  - If all (or first?) records have a field with the value of id-field-name, then that value is used and id mode is 'field'
  - if the records do not have a field name with value of id-field-name, then the id mode is 'index' (or we throw an error since this is just a bad configuration...)

What's a bummer about the above is what if you have a field named `id` but don't want it to be used (you know the id isn't unique, etc, and want to be in index mode). You can't really express that cleanly (maybe choose a random field name and hope it won't accidentally exist).

I'm on the fence about making the config more magical.

Reply:
I think I'd prefer it to be explicit. Setting the `id-field-name` to `id` isn't difficult or verbose, so I'd rather not try to do anything too fancy in an effort to keep a client from having to set that attribute.